lib/shipitron/client/fetch_clusters.rb

```diff
@@ -0,0 +1,32 @@
+require 'shipitron'
+require 'resolv'
+
+module Shipitron
+  module Client
+    class FetchClusters
+      include Metaractor
+
+      required :cluster_discovery
+
+      def call
+        Resolv::DNS.open do |dns|
+          resources = dns.getresources(
+            context.cluster_discovery,
+            Resolv::DNS::Resource::IN::SRV
+          ).sort_by! do |a|
+            [a.priority, a.weight]
+          end.reverse!
+
+          context.clusters = resources.map do |r|
+            Smash.new(
```

Comment:
I can't find code to look at for this gem.

Reply:
It's in shipitron.  See smash.rb.

Reply:
Note: newer hashie has support for this as a mixin

Reply:
Ah ok, cause there is a gem but they don't point to GitHub... so I was like :what: